### PR TITLE
TS types

### PR DESCRIPTION
**Pull request by kibertoad**
Second iteration of #43, supporting named imports as well. Note that this still doesn't support `import SonicBoom = require('sonic-boom');` style of imports the way DefinitelyTyped types did. See https://github.com/fastify/fastify/issues/2404 and https://github.com/fastify/fastify-swagger/pull/282 for in-depth discussion on the topic of why organize exports this way. If we used `export = SonicBoom;` as it was the case in the past, neither named, nor default imports would work without `esModuleInterop` enabled.

@mcollina If we want to avoid semver major, I can create a separate PR that includes DefinitelyTyped types verbatim, avoiding any incompatibilities, but not playing nice with proper ESM imports, and then create a separate semver-major one that changes the export signature. Your call.

### CODE REVIEW

**Comment by mcollina** [comment]
I'll just go with the semver-major route when we ship pino v7.

Also, I think we'd need https://github.com/fastify/fastify/blob/master/fastify.js#L600-L613 as well.

cc @fox1t

**Comment by kibertoad** [comment]
okay! I'll add them

**Comment by fox1t** [comment]
@mcollina yes we need them since https://github.com/mcollina/sonic-boom/pull/45/files#diff-88bec6beae84369c0376b56b3bb88fe1R8 exports default and https://github.com/mcollina/sonic-boom/pull/45/files#diff-88bec6beae84369c0376b56b3bb88fe1R10 exports named.

@kibertoad good work!

**Review by fox1t — COMMENTED** [review]
After adding the "infamous triplet" this can be merged!

File: types/index.test-d.ts [hunk 1]
```diff
@@ -1,10 +1,18 @@
-import { SonicBoom } from "../";
-import SonicBoomEsm from "../";
+import { SonicBoom as SonicBoomNamed } from "../";
+import SonicBoomDefault from "../";
```
Comment (fox1t) [inline types/index.test-d.ts:2]: Nice! We should add this import test in every package we maintain :)

**Review by fox1t — APPROVED** [review]
LGTM

**Comment by mcollina** [comment]
@kibertoad could you rebase this PR?

**Comment by kibertoad** [comment]
@mcollina Sure thing, will focus on everything needed for pino types this evening.

**Comment by kibertoad** [comment]
@mcollina Done!

**Review by mcollina — APPROVED** [review]
lgtm

**Pull request merged**